Generated C in lib test triggers non-actionable signed-literal warning in temp test.c

### Summary
`bosatsu lib test` emits a C compiler warning from generated temporary C code (`test.c`) about a too-large signed integer literal, which is noisy and not actionable for user Bosatsu source.

### Minimal reproduction
Run:
```bash
bosatsu lib test --name mylib --filter 'MyLib/Fib'
```

### Actual output
```text
.../test.c:14326:81: warning: integer literal is too large to be represented in a signed integer type, interpreting as unsigned [-Wimplicitly-unsigned-literal]
  ... bsts_float64_from_bits(-9223372036854775808)),
```

### Expected output
Generated C should avoid this warning (for example by using an unsigned constant form/cast that is warning-free on clang), or warnings from generated internals should be suppressed/mapped so users are not exposed to non-actionable diagnostics.

### Environment
- bosatsu: `0.0.36`
- OS: macOS arm64 (Darwin 25.3.0)

--- a/core/src/main/scala/dev/bosatsu/codegen/clang/ClangGen.scala
+++ b/core/src/main/scala/dev/bosatsu/codegen/clang/ClangGen.scala
@@ -550,9 +550,10 @@ class ClangGen[K](ns: CompilationNamespace[K]) {
 
           case Lit.Str(toStr) => StringApi.fromString(toStr)
           case f: Lit.Float64 =>
+            val rawBits = BigInt(java.lang.Long.toUnsignedString(f.toRawLongBits))
             pv(
               Code.Ident("bsts_float64_from_bits")(
-                Code.IntLiteral(BigInt(f.toRawLongBits))
+                Code.Ident("UINT64_C")(Code.IntLiteral(rawBits))
               )
             )
         }

--- a/core/src/test/scala/dev/bosatsu/codegen/clang/ClangGenTest.scala
+++ b/core/src/test/scala/dev/bosatsu/codegen/clang/ClangGenTest.scala
@@ -450,6 +450,27 @@ main = 1.5
     }
   }
 
+  test("float literals with sign bit use unsigned bit literals") {
+    TestUtils.checkPackageMap("""
+main = -0.0
+""") { pm =>
+      val renderedE = Par.withEC {
+        ClangGen(pm).renderMain(
+          TestUtils.testPackage,
+          Identifier.Name("main"),
+          Code.Ident("run_main")
+        )
+      }
+      renderedE match {
+        case Left(err) =>
+          fail(err.toString)
+        case Right(doc) =>
+          val rendered = doc.render(80)
+          assert(rendered.contains("UINT64_C(9223372036854775808)"))
+      }
+    }
+  }
+
   test("float literal pattern matching uses float equality helper") {
     TestUtils.checkPackageMap("""
 def is_one(x):

